chef/automate
# [infra-proxy-service] User management - GET users list API

### :nut_and_bolt: Description: What code changed, and why?
This PR is about listing the users in an organisation
### :chains: Related Resources
https://github.com/chef/automate/issues/5477
### :+1: Definition of Done
I have added API to list the users in an organisation
### :athletic_shoe: How to Build and Test the Change
Steps to build:

```
rebuild components/infra-proxy-service
rebuild components/automate-gateway
```
To create and add user to an organisation, you can use below steps
1. Ensure Chef Infra Server is running. If it is not start it with ```start_chef_server```
2. To create user, use ```chef-server-ctl user-create USER_NAME FIRST_NAME [MIDDLE_NAME] LAST_NAME EMAIL 
    'PASSWORD' (options)```
3. Verify the created user with ```chef-server-ctl user-list```
4. Then add a user to an organisation with ```chef-server-ctl org-user-add ORG_NAME USER_NAME (options)```

Steps to test:

1. Ensure Chef Infra Server is running. If it is not start it with ```start_chef_server```
2. Add the chef-server test environment credentials and extract the SERVER and ORG IDs.

Request
```
curl -sSkH "api-token: $(get_admin_token)" 'https://a2-dev.test/api/v0/infra/servers/SERVER_ID/orgs/ORG_ID/users'
```
Response
```
{
    "users": [
        {
            "username": "himanshi-test1"
        },
        {
            "username": "sonali-test"
        },
        {
            "username": "vinay-test"
        }
    ]
}
```
### :white_check_mark: Checklist

**All PRs** must tick these:


With occasional exceptions, all PRs **from Progress employees** must tick these:


All PRs **from Progress employees** should tick these if appropriate:


*Please add a note next to any checkbox above if you are NOT ticking it.*

### :camera: Screenshots, if applicable
